observer.go

```diff
@@ -107,5 +110,17 @@ func Handle[T any](ctx context.Context, o Observable[T], h func(context.Context,
 				}
 			}
 		}
-	}, unsub
+	}
+}
+
+// Close disconnects all clients from the observer
+func (o *Observer[T]) Close() error {
+	var res error
+	o.clients.Range(func(key, _ any) bool {
+		if err := o.deleteClient(key); err != nil {
+			res = multierror.Append(res, err)
```

Comment:
Shouldn't it be enough to just return the error? This adds another dependency to the library which is not really necessary in my opinion. If you have a good reason for this, please enlighten me